File: 496-next-greater-element-i/next-greater-element-i.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        stack=[]
        nextgreater={}
        for num in nums2:
            while stack!=[] and num>stack[-1]:
                nextgreater[stack.pop()]=num
            stack.append(num)
        
        while stack!=[]:
            nextgreater[stack.pop()]=-1
        
        res=[]
        for num in nums1:
            res.append(nextgreater[num])
        return res

File: 496-next-greater-element-i/next-greater-element-i.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {

        HashMap<Integer,Integer> map=new HashMap<>();
        Stack<Integer> stack=new Stack<>();


        for(int num: nums2){
            while(stack.isEmpty()==false && num>stack.peek()){
                map.put(stack.pop(),num);
            }
            stack.push(num);
        }

        while(stack.isEmpty()==false){
            map.put(stack.pop(),-1);
        }

        int[] res=new int[nums1.length];
        int index=0;
        for(int num: nums1){
            res[index]=map.get(num);
            index++;
        }

        return res;


    }
}